src/flows/pipes/RemoteCSV/view.tsx

```diff
@@ -0,0 +1,94 @@
+import React, {FC, useState, useContext} from 'react'
+
+import {
+  Dropdown,
+  ComponentColor,
+  Input,
+  InputType,
+  ComponentStatus,
+} from '@influxdata/clockface'
+import {PipeProp} from 'src/types/flows'
+import {PipeContext} from 'src/flows/context/pipe'
+
+const SampleCSVs = {
+  Sample_Air_Sensor_Data:
+    'https://raw.githubusercontent.com/influxdata/influxdb2-sample-data/master/air-sensor-data/air-sensor-data-annotated.csv',
+  NOAA_Weather_Data: 'https://influx-testdata.s3.amazonaws.com/noaa.csv',
+  USGS_Earthquake_Data:
+    'https://raw.githubusercontent.com/influxdata/influxdb2-sample-data/master/usgs-earthquake-data/all_week-annotated.csv',
+  'NOAA_National_Buoy_Data_Center_(NDBC)':
+    'https://raw.githubusercontent.com/influxdata/influxdb2-sample-data/master/noaa-ndbc-data/latest-observations-annotated.csv',
+  Custom: '',
+}
+
+const RemoteCSV: FC<PipeProp> = ({Context}) => {
+  const {data, update} = useContext(PipeContext)
+  const [selectedCSV, setSelectedCSV] = useState(data.csvType ?? '')
```

Comment:
dont cache like this. just do a straight functional transform off of data and use update to... update the data